R/tpar.R

```diff
@@ -8,274 +8,325 @@
+      stop("facet.bg needs to be NULL, or a numeric or character", call. = FALSE)
+    }
+    assert_true(length(facet.bg) == 1, name = "length(facet.bg)==1")
+  }
+
+  facet.border = .tpar$facet.border
+  if (!is.null(facet.border)) {
+    if (!is.numeric(facet.border) && !is.character(facet.border) && !is.na(facet.border)) {
+      stop("facet.border needs to be NULL, or a numeric, character, or NA", call. = FALSE)
+    }
+    assert_true(length(facet.border) == 1, name = "length(facet.border)==1")
+  }
+}
+
+
+init_tpar = function(rm_hook = FALSE) {
+  rm(list = names(.tpar), envir = .tpar)
+
+  if (isTRUE(rm_hook)) {
+    hook = getHook("before.plot.new")
+    if (length(hook) > 0) {
+      # need weird function because of Quarto evaluate::evaluate failure
+      # setHook("before.plot.new", NULL, action = "replace")
+      setHook("before.plot.new", function() NULL, action = "replace")
+    }
+  }
+
+  # Figure output options if written to file
+  .tpar$file.width = if (is.null(getOption("tinyplot_file.width"))) 7 else as.numeric(getOption("tinyplot_file.width"))
+  .tpar$file.height = if (is.null(getOption("tinyplot_file.height"))) 7 else as.numeric(getOption("tinyplot_file.height"))
+  .tpar$file.res = if (is.null(getOption("tinyplot_file.res"))) 300 else as.numeric(getOption("tinyplot_file.res"))
+
+  # Facet margin, i.e. gap between the individual facet windows
+  .tpar$fmar = if (is.null(getOption("tinyplot_fmar"))) c(1, 1, 1, 1) else as.numeric(getOption("tinyplot_fmar"))
+
+  # Other facet options
+  .tpar$facet.cex = if (is.null(getOption("tinyplot_facet.cex"))) 1 else as.numeric(getOption("tinyplot_facet.cex"))
+  .tpar$facet.font = if (is.null(getOption("tinyplot_facet.font"))) NULL else as.numeric(getOption("tinyplot_facet.font"))
+  .tpar$facet.col = if (is.null(getOption("tinyplot_facet.col"))) NULL else getOption("tinyplot_facet.col")
+  .tpar$facet.bg = if (is.null(getOption("tinyplot_facet.bg"))) NULL else getOption("tinyplot_facet.bg")
+  .tpar$facet.border = if (is.null(getOption("tinyplot_facet.border"))) NA else getOption("tinyplot_facet.border")
+
+  # Plot grid
+  .tpar$grid = if (is.null(getOption("tinyplot_grid"))) FALSE else as.logical(getOption("tinyplot_grid"))
+  .tpar$grid.col = if (is.null(getOption("tinyplot_grid.col"))) "lightgray" else getOption("tinyplot_grid.col")
+  .tpar$grid.lty = if (is.null(getOption("tinyplot_grid.lty"))) "dotted" else getOption("tinyplot_grid.lty")
+  .tpar$grid.lwd = if (is.null(getOption("tinyplot_grid.lwd"))) 1 else as.numeric(getOption("tinyplot_grid.lwd"))
+
+  # Legend margin, i.e. gap between the legend and the plot elements
+  .tpar$lmar = if (is.null(getOption("tinyplot_lmar"))) c(1.0, 0.1) else as.numeric(getOption("tinyplot_lmar"))
+
+  # Alpha fill (transparency) default for ribbon and area plots
+  .tpar$ribbon.alpha = if (is.null(getOption("tinyplot_ribbon.alpha"))) 0.2 else as.numeric(getOption("tinyplot_ribbon.alpha"))
+}
```

Comment:
See my other comment about whether these should be kept in zzz.R and made available onLoad.

Reply:
https://github.com/grantmcdermott/tinyplot/pull/258#discussion_r1861181158